experiments/dask/bench.yml

```diff
@@ -0,0 +1,104 @@
+# TODO
+#  * add support for item.repetitions
+- name: pull image
+  shell: docker pull {{ item.image_name }}
+- name: pull alpine
+  shell: docker pull alpine
+
+- name: drop caches
+  shell: echo 3 > /proc/sys/vm/drop_caches
+
+- name: run benchmark on localfs
+  shell: >
+    docker run --rm \
+      -v {{ item.folder_name }}:/mount \
+      {{ item.docker_flags }} \
+      {{ item.image_name }} {{ item.image_args }}
+  when: "{{ item.run_in_local_fs | bool }}"
+  register: docker_output
+
+- name: write results to output
+  shell: echo "{{ item.id }}, exec, local, {{docker_output.delta}}" >> results.csv
+  when: "{{ item.run_in_local_fs | bool }}"
+
+# gassyfs
+
+- name: copy folder into gassyfs
+  shell: >
+    docker run --rm \
+      --volumes-from gassyfs-master \
```

Comment:
In my experience, the `docker` module is usually behind `docker-py` which
in turn is usually behind `docker`, so I stopped using it altogether .
Also, the `docker` module requires `docker-py` on the remote nodes, which
is not always present

On Fri, Apr 8, 2016, 07:43 Michael Sevilla <notifications@github.com> wrote:

> In experiments/dask/bench.yml
> <https://github.com/michaelsevilla/sc16/pull/9#discussion_r59035166>:
>
> > +      -v {{ item.folder_name }}:/mount \
> > +      {{ item.docker_flags }} \
> > +      {{ item.image_name }} {{ item.image_args }}
> > +  when: "{{ item.run_in_local_fs | bool }}"
> > +  register: docker_output
> > +
> > +- name: write results to output
> > +  shell: echo "{{ item.id }}, exec, local, {{docker_output.delta}}" >> results.csv
> > +  when: "{{ item.run_in_local_fs | bool }}"
> > +
> > +# gassyfs
> > +
> > +- name: copy folder into gassyfs
> > +  shell: >
> > +    docker run --rm \
> > +      --volumes-from gassyfs-master \
>
> Why do you use 'shell docker run blah'? Is it because Ansible doesn't
> support '--volumes-from'?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/michaelsevilla/sc16/pull/9/files/93d172537fe6f305ffd2825f512dc211354ca627#r59035166>
>